python/cog/server/worker.py

```diff
@@ -215,10 +219,14 @@ def _consume_events_inner(self) -> None:
 
             self._predict_payload = None
             self._predict_start.clear()
-            self._result.set_result(done)
+
+            # We capture the predict future and then set state to READY before
+            # completing it, so that we can immediately accept work.
+            result = self._result
```

Comment:
I know it's the same in practice, but it feels safer to couple clearing `self._predict_start` with the rest of resetting the state of the worker:

```suggestion
            # We capture the predict future and then set state to READY before
            # completing it, so that we can immediately accept work.
            result = self._result
            self._predict_payload = None
            self._predict_start.clear()
```